tutorials/how-to-implement-rag-generativeapis/index.mdx

```diff
@@ -70,288 +76,275 @@ Create a .env file and add the following variables. These will store your API ke
+           )
+   
+   prompt = hub.pull("rlm/rag-prompt")
+   retriever = vector_store.as_retriever()
+   
+   rag_chain = (
+           {"context": retriever, "question": RunnablePassthrough()}
+           | prompt
+           | llm
+           | StrOutputParser()
+       )
+   
+   for r in rag_chain.stream("Provide the CLI command to shut down a scaleway instance. Its instance-uuid is example-28f3-4e91-b2af-4c3502562d72"):
+       print(r, end="", flush=True)
+   ```
+   - `hub.pull("rlm/rag-prompt")` uses a standard RAG template. This ensures documents content retrieved will be passed as context along your prompt to the LLM using a compatible format.
+   - `vector_store.as_retriever()` configures your vector store as additional context to retrieve before calling the LLM. By default, the 4 closest document chunks are retrieved using vector `similarity` score.
+   - `rag_chain` defines a workflow performing the following steps in order: Retrieve relevant documents, Prompt LLM with document as context, and final output parsing.
+   - `for r in rag_chain.stream("Prompt question")` starts the rag workflow with `Prompt question` as input.
 
-prompt = hub.pull("rlm/rag-prompt")
-retriever = vector_store.as_retriever()
+4. You can now execute your RAG pipeline with:
 
+   ```sh
+   python rag.py
+   ```
 
-rag_chain = (
-        {"context": retriever, "question": RunnablePassthrough()}
-        | prompt
-        | llm
-        | StrOutputParser()
-    )
+   If you used the Scaleway cheatsheet provided as examples, and asked for a CLI command to power of instance, you should see the following answer:
+   ```sh
+   scw instance server stop example-28f3-4e91-b2af-4c3502562d72 
 
-for r in rag_chain.stream("Your question"):
-    print(r, end="", flush=True)
-    time.sleep(0.1)
-```
-- LLM initialization: we initialize the ChatOpenAI instance using the endpoint and API key from the environment variables, along with the specified model name.
+   This will shut down the instance with the specified instance-uuid. 
+   Please note that this command only stops the instance, it doesn't shut it down completely
+   ```
+   This command is fully correct and can be used with Scaleway CLI. Note especially that vector embedding enabled the system to retrieve proper document chunks even if the Scaleway cheatsheet never mention `shut down` but only `power off`.
+   You can compare this result without RAG (for instance by using the same prompt in [Generative APIs Playground](https://console.scaleway.com/generative-api/models/fr-par/playground?modelName=llama-3.1-8b-instruct)):
+   ```sh
+   scaleway instance shutdown --instance-uuid example-28f3-4e91-b2af-4c3502562d72
+   ```
+   This is command is not correct at all, and "hallucinate" in several ways to fit the question prompt content: `scaleway` instead of `scw`, `instance` instead of `instance server`, `shutdown` instead of `stop` and `--instance-uuid` parameter doesn't exist.
```

Comment:
```suggestion

   This command is incorrect and 'hallucinates' in several ways to fit the question prompt content: scaleway instead of scw, instance instead of instance server, shutdown instead of stop, and the --instance-uuid parameter does not exist.
```